content/en/docs/collector/building/authenticator-extension.md

```diff
@@ -122,41 +122,41 @@ service:
 ### Server authenticators
 
 A [server authenticator][sa] is essentially an extension with an `Authenticate`
-function, receiving the payload headers as parameter. If the authenticator is
-able to authenticate the incoming connection, it should return a `nil` error, or
-the concrete error if it can't. As an extension, the authenticator should make
-sure to initialize all the resources it needs during the
+function, receiving the payload headers as a parameter. If the authenticator can
+authenticate the incoming connection, it should return a `nil` error, or the
+concrete error if it can't. As an extension, the authenticator should ensure to
+initialize all the resources it needs during the
 [`Start`](https://pkg.go.dev/go.opentelemetry.io/collector/component#Component)
-phase, and is expected to clean them up upon `Shutdown`.
+phase, and it is expected to clean them up upon `Shutdown`.
 
 The `Authenticate` call is part of the hot path for incoming requests and will
-block the pipeline, so make sure to properly handle any blocking operations you
-need to make. Concretely, respect the deadline set by the context, in case one
-is provided. Also make sure to add enough observability to your extension,
-especially in the form of metrics and traces, so that users can get setup a
-notification system in case error rates go up beyond a certain level and can
-debug specific failures.
+block the pipeline. Therefore, make sure to handle any blocking operations you
+need to make properly. Specifically, respect the deadline set by the context, if
+one is provided. Also, make sure to add enough observability to your extension,
+especially in the form of metrics and traces. This will enable users to set up a
+notification system that alerts them in case error rates exceed a certain level,
+as well as to debug specific failures.
```

Comment:
```suggestion
this function finishes. Because of that, your authenticator must avoid slow or unnecessary blocking work. If the `context` sets a deadline, make sure your code follows it so the pipeline isn't delayed or left hanging.

You should also add good observability to your authenticator, especially metrics and traces. This helps users set up alerts if errors start increasing and makes it easier for them to troubleshoot authentication problems.
```